lib/src/native/realm_core.dart

```diff
@@ -196,6 +198,28 @@ class _RealmCore {
     });
   }
 
+  static int initial_data_callback(Object configHandle, Pointer<shared_realm> realmHandle) {
+    try {
+      final config = configHandle as Configuration;
+      final realm = RealmInternal.getUnowned(config, RealmHandle._unowned(realmHandle));
```

Comment:
A bit unintuitive you need to guard against `configHandle` being null. Has to with the way `Dart_HandleFromWeakPersistent_DL` works, and the fact that Google in their infinite wisdom, has not exposed `Dart_IsNull_DL` (https://github.com/dart-lang/sdk/issues/48331)

Reply:
If you do a `if (config is Configuration)` you are golden